### PR TITLE
fix/resources-release

### PR DESCRIPTION
**Pull request by PaulDalek**
Added fixes and corrections to releasing pool and resources, closing servers and clearing intervals.

### CODE REVIEW

File: lib/server/server.js [hunk 1]
```diff
@@ -100,11 +109,14 @@ export const startServer = async (serverConfig) => {
...
 
       // Attach error handlers and listen to the server
-      attachErrorHandlers(httpServer);
+      attachServerErrorHandlers(httpServer);
 
       // Listen
       httpServer.listen(serverConfig.port, serverConfig.host);
 
+      // Save the reference to HTTP server
+      activeServers.push(httpServer);
```
Comment (jszuminski) [inline lib/server/server.js:118]: I'm not sure if that'll ever be the case (maybe I'm missing something here), but I see that we do `activeServers.push(httpServer)`, but we never clean the `activeServers`. Even `server.on('close', () => ...)` we only log that the server has been closed. Maybe if there are multiple active servers and one of them gets closed we should remove it from the array?

Comment (PaulDalek) [reply lib/server/server.js:118]: @jszuminski Actually, we do close servers. Take a look at the `shutdownCleanUp` function, where servers are get with the `getServers`.

**Review by jszuminski — COMMENTED** [review]
I've noticed that when running `npm run node-tests` I get the following error:
```
[Fail] Node module from file: options_stringified.json, took: 133ms.
Mon Apr 22 2024 10:21:12 GMT+0200 [error] - Cannot create property 'chart' on string '{"title":{"text":"Stringified options"},"xAxis":{"categories":["Jan","Feb","Mar","Apr","May","Jun","Jul","Aug","Sep","Oct","Nov","Dec"]},"series":[{"type":"column","data":[1,3,2,4]},{"type":"column","data":[5,3,4,2]}]}' 
 TypeError: Cannot create property 'chart' on string '{"title":{"text":"Stringified options"},"xAxis":{"categories":["Jan","Feb","Mar","Apr","May","Jun","Jul","Aug","Sep","Oct","Nov","Dec"]},"series":[{"type":"column","data":[1,3,2,4]},{"type":"column","data":[5,3,4,2]}]}'
    at doExport (file:///Users/jakubszuminski/Documents/Coding/Highcharts/exporting/node-export-server/lib/chart.js:360:21)
    at Object.startExport (file:///Users/jakubszuminski/Documents/Coding/Highcharts/exporting/node-export-server/lib/chart.js:108:11)
    at file:///Users/jakubszuminski/Documents/Coding/Highcharts/exporting/node-export-server/tests/node/node_test_runner.js:95:18
    at new Promise (<anonymous>)
    at file:///Users/jakubszuminski/Documents/Coding/Highcharts/exporting/node-export-server/tests/node/node_test_runner.js:72:13
    at Array.map (<anonymous>)
    at file:///Users/jakubszuminski/Documents/Coding/Highcharts/exporting/node-export-server/tests/node/node_test_runner.js:70:10
```
but I've checked and it appears on master as well so this is not a part of this pull request.

**Review by jszuminski — APPROVED** [review]
Looking great!

**Comment by PaulDalek** [comment]
@jszuminski Thanks for the reveiw and noticing the problem with the test. Will investigate it separately and fix it.

**Comment by mikemeerschaert** [comment]
FYI this PR appears to remove the `killPool` function. I'm not sure if this was intentional or is supposed to be replaced by the `shutdownCleanUp` function, but if that is the case, it means you can't clean up the highcharts pool while still executing code in a node module since it causes the whole process to exit.

**Comment by PaulDalek** [comment]
@mikemeerschaert Thanks for noticing that! The `initPool` and `killPool` are available again.

**Pull request merged**